client/tasks/tasks.tsx

```diff
@@ -1,3 +1,146 @@
-export const Tasks: React.FC = () => (
-	<div>Task lists will be rendered here.</div>
-);
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { MenuGroup, MenuItem } from '@wordpress/components';
+import { check } from '@wordpress/icons';
+import { Fragment, useEffect } from '@wordpress/element';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { ONBOARDING_STORE_NAME, OPTIONS_STORE_NAME } from '@woocommerce/data';
+import { useExperiment } from '@woocommerce/explat';
+import { recordEvent } from '@woocommerce/tracks';
+
+/**
+ * Internal dependencies
+ */
+import { DisplayOption } from '../header/activity-panel/display-options';
+import { Task } from './task';
+import { TaskList } from './task-list';
+import { TasksPlaceholder } from './placeholder';
+
+export type TasksProps = {
+	query: { task: string };
+};
+
+export const Tasks: React.FC< TasksProps > = ( { query } ) => {
+	const { task } = query;
+	const { hideTaskList } = useDispatch( ONBOARDING_STORE_NAME );
+	const { updateOptions } = useDispatch( OPTIONS_STORE_NAME );
+	const [ isLoadingExperiment, experimentAssignment ] = useExperiment(
+		'woocommerce_tasklist_progression'
+	);
+
+	const { isResolving, taskLists } = useSelect( ( select ) => {
+		return {
+			isResolving: select( ONBOARDING_STORE_NAME ).isResolving(
+				'getTasks'
+			),
+			taskLists: select( ONBOARDING_STORE_NAME ).getTaskLists(),
```

Comment:
It looks like we have a type definition for this in `task-list.tsx`. Would it be useful to centralize these with a types file and use it here?

Reply:
For sharing the type definitions between the data store and the components?  I wasn't sure if we could do this since the data store is in a separate package.

Do we have an example somewhere of this?

Reply:
I haven't looked deeply into this, but it [looks like we're exporting some types from the data store](https://github.com/woocommerce/woocommerce-admin/blob/a29e67554ac85589fa3c773e2afa4e0416f2e56e/packages/data/src/index.ts?plain=1#L23).

Reply:
I've actually been seeing a number of similar linting issues, but I didn't thin they were consequential? Hadn't tested that theory.